misc/python/materialize/cli/ci_coverage_pr_report.py

```diff
@@ -0,0 +1,90 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import re
+import subprocess
+import sys
+from collections import OrderedDict
+from itertools import groupby
+from typing import Dict, List
+
+Coverage = Dict[str, OrderedDict[int, int]]
+SOURCE_RE = re.compile(
+    "^/var/lib/buildkite-agent/builds/buildkite-.*/materialize/tests/(.*$)"
+)
+
+
+def find_interesting_lines() -> Coverage:
+    coverage: Coverage = {}
+    # TODO: Get base state from buildkite instead of just HEAD~
+    result = subprocess.run(
+        ["git", "diff", "-U0", "HEAD~"], check=True, capture_output=True
+    )
+    file = None
+    for line_raw in result.stdout.splitlines():
+        line = line_raw.decode("utf-8")
+        if line.startswith("+++"):
+            if not line.endswith(".rs"):
+                continue
+            file = line.removeprefix("+++ b/")
+            coverage[file] = OrderedDict()
+        elif line.startswith("@@ ") and file:
+            parts = line.split(" ")[2]
```

Comment:
a comment here that you are zeroing in on the second argument to `@@` , the lines added, would be nice:

```
@@ -527 +526,0 @@ fn parse_kafka_opt(opt: &str) -> (String, String) {
```